test/dart3_suggestors/null_safety_prep/fn_prop_null_aware_call_suggestor_test.dart

```diff
@@ -0,0 +1,225 @@
+                  
+                  return (Dom.button()..onClick = handleClick)();
+                },
+                UiFactoryConfig(displayName: 'Foo'),
+              );
+            '''),
+            expectedOutput: withOverReactImport('''
+              final Foo = uiFunction<UiProps>(
+                (props) {
+                  final handleClick = useCallback<MouseEventCallback>((e) {
+                    props.onClick?.call(e);
+                  }, [props.onClick]);
+                  
+                  return (Dom.button()..onClick = handleClick)();
+                },
+                UiFactoryConfig(displayName: 'Foo'),
+              );
+            '''));
+      });
+
+      test(
+          'unless the single condition is not a null check of the function being called',
+          () async {
+        await testSuggestor(
+          expectedPatchCount: 0,
+          input: withOverReactImport('''
+              final Foo = uiFunction<UiProps>(
+                (props) {
+                  final handleClick = useCallback<MouseEventCallback>((e) {
+                    if (1 > 0) {
+                      props.onClick(e);
+                    }
+                  }, [props.onClick]);
+                  
+                  return (Dom.button()..onClick = handleClick)();
+                },
+                UiFactoryConfig(displayName: 'Foo'),
+              );
+            '''),
+        );
+      });
+
+      test(
+          'unless the single condition involves the function being called, but is not a null check',
+              () async {
+            await testSuggestor(
+              expectedPatchCount: 0,
+              input: withOverReactImport('''
+              final Foo = uiFunction<UiProps>(
+                (props) {
+                  final handleClick = useCallback<MouseEventCallback>((e) {
+                    if (props.onClick is Function) {
+                      props.onClick(e);
+                    }
+                  }, [props.onClick]);
+                  
+                  return (Dom.button()..onClick = handleClick)();
+                },
+                UiFactoryConfig(displayName: 'Foo'),
+              );
+            '''),
+            );
+          });
+
+      test('unless there are multiple conditions', () async {
+        await testSuggestor(
+          expectedPatchCount: 0,
+          input: withOverReactImport('''
+              final Foo = uiFunction<UiProps>(
+                (props) {
+                  final handleClick = useCallback<MouseEventCallback>((e) {
+                    if (props.onClick != null && props.onMouseEnter != null) {
+                      props.onClick(e);
+                    }
+                  }, [props.onClick]);
+                  
+                  return (Dom.button()..onClick = handleClick)();
+                },
+                UiFactoryConfig(displayName: 'Foo'),
+              );
+            '''),
+        );
+      });
+    });
```

Comment:
Could we also add test cases that
- if-checks that don't involve props aren't updated (and the codemod skips over them without throwing)
- then statements that `return` something aren't updated
- then statements that do something besides call the prop aren't updated